mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext/src/main/java/org/eclipse/mylyn/wikitext/util/IDStrategies.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
```

Comment:
For what it's worth, Github Copilot came up with this solution:

```
			return id.toLowerCase(Locale.ENGLISH)
				.replaceAll("[^a-z0-9_]", "-")
				.replaceAll("-+", "-")
				.replaceAll("^-|-$", "");
```

Reply:
Looks nice :)

Could performance be an issue with all those regular expressions (patterns compiled anew for each invocation)?